frameworks/hdfs/docs/security.md

```diff
@@ -0,0 +1,184 @@
+HTTP/name-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-0-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/name-0-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/name-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/name-1-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/name-1-zkfc.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+
+example/journal-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/journal-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/journal-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/journal-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/journal-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/journal-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+
+example/data-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/data-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/data-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/data-1-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+example/data-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+HTTP/data-2-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE
+```
+
+{% include services/security-kerberos-ad.md
+    principal="example/name-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE"
+    spn="example/name-0-node.agoodexample.autoip.dcos.thisdcos.directory"
+    upn="example/name-0-node.agoodexample.autoip.dcos.thisdcos.directory@EXAMPLE" %}
+
+{% include services/security-service-keytab.md
+    techName="Apache HDFS" %}
+
+#### Install the Service
+
+Install the DC/OS Apache HDFS service with the following options in addition to your own:
+```json
+{
+    "service": {
+        "kerberos": {
+            "enabled": true,
+            "kdc": {
+                "hostname": "<kdc host>",
+                "port": <kdc port>
+            },
+            "primary": "<service primary default hdfs>",
+            "realm": "<realm>",
+            "keytab_secret": "<path to keytab secret>",
+            "debug": <true|false default false>
+        }
+    }
+}
+```
+
+<!-- TO BE DETERMINED *Note*: It is possible to enable Kerberos after initial installation but the service may be unavailable during the transition. Additionally, your HDFS clients will need to be reconfigured. -->
+
+## Authorization
+
+The DC/OS Apache HDFS service supports HDFS's native authorization, which behaves similarly to UNIX file permissions. If Keberos is enabled as detailed [above](#kerberos-authentication), then Kerberos principals are mapped to HDFS users against which permissions can be assigned.
+
+### Enable Authorization
+
+#### Prerequisites
+- Completion of  [Kerberos authentication](#kerberos-authentication) above.
+
+#### Set Kerberos Principal to User Mapping
```

Comment:
So we're intentionally not providing an example or a link as to how this mapping would work?

Reply:
https://github.com/mesosphere/dcos-commons/pull/2216/files/d9669c2cca0e5bc33fc3b4711d935d214e3c2b1b#diff-874c92dd598ecf9b90ea2d5322ef75a2R182

Reply:
I can't read. Carry on.